protocol/amqp/v2/receiver.go

```diff
@@ -19,10 +19,13 @@ import (
 const serverDown = "session ended by server"
 
 // receiver wraps an amqp.Receiver as a binding.Receiver
-type receiver struct{ amqp *amqp.Receiver }
+type receiver struct {
+	amqp    *amqp.Receiver
+	options *amqp.ReceiveOptions
```

Comment:
question: do we need pointer semantics here (and in the constructors)?

Reply:
if you notice the changes on `sender` file require a pointer since the library used assumes that if nil is passed default is assumed. If in the future the library does the same for the receiver it will be prepared to it. WDYT?

Reply:
That is fine, you can always use the copied value with `&var` - but make a copy of the provided struct (to avoid races/bugs) when calling `New...()`

Reply:
sure, gone back to not using pointer for this case